Update utilities API docs

mdo opened the pull request:
Changes so far:

- Reformats list of utilities options into a responsive table.
- Did some copyediting on body text and headings.
- Cleaned up the formatting of our outputted CSS so it's not as lengthy of a page.

Still to do:

- [x] Nest all headings/sections after `API explained` under that heading
- [x] Add new section for `Using the API` after that, with content from #31328 as a draft
- [ ] Bonus: add an example to the WIP v5 bootstrap-npm-starter project

/cc @MartijnCuppens @XhmikosR 

Preview: <https://deploy-preview-32049--twbs-bootstrap.netlify.app/docs/5.0/utilities/api/>

Closes #31328 by superseding it

Luke-zhang-04 committed:
docs: add info on modifying and adding utilities

mdo committed:
Editing and formatting utilities API page

XhmikosR reviewed (approved):
LGTM but I'd like a look from @MartijnCuppens

MartijnCuppens commented:
Looks good so far!

mdo committed:
Big copy edits and testing

Moves the remote utilities section to the using the api section and updates the code snippet to work. Previous one removed all utilities.

mdo commented:
Alrighty this is ready to go I think! Tested all the new snippets I added under the `Using the API` section—everything should be good now.

XhmikosR committed:
Merge branch 'main' into utils-api-edits
